File: proyectoweb/src/main/java/co/edu/unicauca/competencias/proyectoweb/Teacher_module/Teacher_infrastructure/controllers/TeacherController.ts
import { Request, Response } from 'express';
import { ITeacherService } from '../../Teacher_service/interfaces/ITeacherService';
import { Teacher } from '../../Teacher_core/entities/Teacher';

export class TeacherController {
  private TeacherService: ITeacherService;

  constructor(TeacherService: ITeacherService) {
    this.TeacherService = TeacherService;
  }

  async getAll(req: Request, res: Response): Promise<void> {
    const Teachers = await this.TeacherService.getAllTeachers();
    res.json(Teachers);
  }

  async getById(req: Request, res: Response): Promise<void> {
    const Teacher = await this.TeacherService.getTeacherById(Number(req.params.id));
    if (Teacher) {
      res.json(Teacher);
    } else {
      res.status(404).json({ message: 'Teacher not found' });
    }
  }

  async create(req: Request, res: Response): Promise<void> {
    const Teacher: Teacher = req.body;
    const newTeacher = await this.TeacherService.createTeacher(Teacher);
    res.status(201).json(newTeacher);
  }

  async update(req: Request, res: Response): Promise<void> {
    const Teacher: Teacher = req.body;
    const updatedTeacher = await this.TeacherService.updateTeacher(Number(req.params.id), Teacher);
    if (updatedTeacher) {
      res.json(updatedTeacher);
    } else {
      res.status(404).json({ message: 'Teacher not found' });
    }
  }

  async delete(req: Request, res: Response): Promise<void> {
    const success = await this.TeacherService.deleteTeacher(Number(req.params.id));
    if (success) {
      res.status(204).send();
    } else {
      res.status(404).json({ message: 'Teacher not found' });
    }
  }
}

File: proyectoweb/src/main/java/co/edu/unicauca/competencias/proyectoweb/Teacher_module/Teacher_infrastructure/controllers/TeacherController.java
package co.edu.unicauca.competencias.proyectoweb.Teacher_module.Teacher_infrastructure.controllers;

import java.util.Optional;

import co.edu.unicauca.competencias.proyectoweb.Teacher_module.Teacher_infrastructure.persistence.DTO.TeacherDTO;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import co.edu.unicauca.competencias.proyectoweb.Teacher_module.Teacher_service.interfaces.ITeacherService;
import org.springframework.http.HttpStatus;


import java.util.List;

@RestController
@RequestMapping("/api/teachers")
@CrossOrigin(origins = "*",maxAge = 3600)
public class TeacherController {
    private final ITeacherService teacherService;

    public TeacherController(ITeacherService teacherService) {
        this.teacherService = teacherService;
    }

    @GetMapping()
    public ResponseEntity<List<TeacherDTO>> getAllTeachers() {
        return ResponseEntity.ok(teacherService.getAllTeachers());
    }

    @GetMapping("/active")
    public ResponseEntity<List<TeacherDTO>> getAllActiveTeachers() {
        return ResponseEntity.ok(teacherService.getAllActiveTeachers());
    }

    @GetMapping("/{id}")
    public ResponseEntity<TeacherDTO> getTeacherById(@PathVariable Integer id) {
        return ResponseEntity.of(teacherService.getTeacherById(id));
    }

    @GetMapping("/name/{names}")
    public ResponseEntity<TeacherDTO> getTeacherByName(@PathVariable String names) {
        return ResponseEntity.of(teacherService.getTeacherByName(names));
    }

    @PostMapping()
    public ResponseEntity<Optional<TeacherDTO>> createTeacher(@RequestBody TeacherDTO teacher) {
        return ResponseEntity.ok(teacherService.createTeacher(teacher));
    }

    @PutMapping("/{id}")
    public ResponseEntity<Optional<TeacherDTO>> updateTeacher(@PathVariable Integer id, @RequestBody TeacherDTO teacher) {
            return ResponseEntity.ok(teacherService.updateTeacher(id, teacher));
    }

    @PutMapping("/status/{id}/{status}")
    public ResponseEntity<Integer> changeStatus(@PathVariable Integer id, @PathVariable String status) {
        return ResponseEntity.ok(teacherService.changeStatus(id, status));
    }
}